client/protocol.js

```diff
@@ -0,0 +1,140 @@
+(function() {
+    'use strict';
+    var Game = window.Game = window.Game || {};
+
+    //
+    // Game protocol implementation
+    //
+    // Game.Socket uses Transport to exchange messages with the server, and then
+    // dispatches them as events (with the ID being the same as in the spec).
+    // TODO validation maybe; could use JSON Schema also
+    //
+    // Actual handling logic is in game.js.
+    //
+
+    Game.Socket = function(transport) {
+        var self = this;
+        var messages = {};
+
+        messages.welcome = function(msg) {
```

Comment:
I'm a spot rusty on JS (and a spot Js on Rust, ... sorry) but yeah, the idea is that when these methods are called, it's a adhoc object that is passed in, we don't need to define some type or explicitly pass each variable, right?